src/strategies/poap-with-weight-v2/index.ts

```diff
@@ -0,0 +1,72 @@
+import { subgraphRequest } from '../../utils';
+import examplesFile from './examples.json';
+
+export const author = 'gawainb';
+export const version = '2.0.0';
+export const examples = examplesFile;
+
+const POAP_API_ENDPOINT_URL = {
+  '1': 'https://api.thegraph.com/subgraphs/name/poap-xyz/poap',
+  '100': 'https://api.thegraph.com/subgraphs/name/poap-xyz/poap-xdai'
+};
+
+const getTokenSupply = {
+  tokens: {
+    __args: {
+      where: {
+        event_: {
+          id_in: undefined
+        }
+      }
+    },
+    event: {
+      id: true,
+      tokenCount: true
+    },
+    id: true,
+    owner: {
+      id: true
+    }
+  }
+};
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  snapshot
+) {
+  // Set eventIds as arguments for GQL query
+  getTokenSupply.tokens.__args.where.event_.id_in = options.eventIds.map(
```

Comment:
Hey @gawainb Everything looks great, thanks for the change, can we add a limit to eventIds? if some one can pass unlimited eventIds it can cause memory issues, throw an error if some one cross the limit,

also subgraph has a limit of `5000`, can respond with it only, better to handle that with a pagination if that can cause some issues :) Thanks

Reply:
Looks like we fixed this issue yesterday.